### PR TITLE
Stabilization -> Development Update 

### PR DESCRIPTION
**Pull request by spham-amzn**
## What does this PR do?

Merges Stabilization/2310(4c271c9) to Development(e703ec8)

## How was this PR tested?

AR

### CODE REVIEW

**Review by amzn-changml — APPROVED** [review]
Approved for sig-build items

**Review by cgalvan — APPROVED** [review]

**Review by moudgils — APPROVED** [review]

File: Code/Framework/AzCore/AzCore/DOM/DomUtils.h [hunk 1]
```diff
@@ -161,49 +161,31 @@ namespace AZ::Dom::Utils
...
         AZStd::any::action_handler_for_t actionHandler;
-
-        if constexpr (AZStd::is_pointer_v<AZStd::remove_cvref_t<WrapperType>>)
+        using ValueType = AZStd::remove_cvref_t<WrapperType>;
+        typeTraits.m_typeId = azrtti_typeid<ValueType>();
+        typeTraits.m_typeSize = sizeof(ValueType);
+
+        // Lifetime variables provides storage for Dom::Value variable
+        // long enough to complete the call to the ValueFromType overload which accepts a void pointer
+        Dom::Value domValueLifetime;
```
Comment (lumberyard-employee-dm) [inline Code/Framework/AzCore/AzCore/DOM/DomUtils.h:170]: The `domValueLifetime` variable can be removed since it is unused.

**Review by lumberyard-employee-dm — APPROVED** [review]

File: Gems/Multiplayer/Code/Include/Multiplayer/AutoGen/AutoComponent_Source.jinja [hunk 2]
```diff
@@ -722,18 +722,10 @@ void {{ ClassName }}::NotifyChanges{{ AutoComponentMacros.GetNetPropertiesSetNam
     }
 {%         endif %}
 {%     else %}
-
```
Comment (AMZN-Gene) [inline Gems/Multiplayer/Code/Include/Multiplayer/AutoGen/AutoComponent_Source.jinja:725]: Oops, this is removing a fix that was added just a few hours ago. We'll want this file to match development (have no changes)
https://github.com/o3de/o3de/commit/1f187c02f4930363973cf4dd9ee85cc48420a545

**Review by AMZN-Gene — CHANGES_REQUESTED** [review]

File: Gems/Multiplayer/Code/Include/Multiplayer/AutoGen/AutoComponent_Source.jinja (hunk 2 above)
Comment (spham-amzn) [reply Gems/Multiplayer/Code/Include/Multiplayer/AutoGen/AutoComponent_Source.jinja]: reverted

**Review by AMZN-Gene — APPROVED** [review]

**Pull request merged**